### PR TITLE
onePingPerDM: add settings

### PR DESCRIPTION
**Pull request by ProffDea**
### Added
- Select option for DM to effect
- Toggle for allowing mentions
- Toggle for allowing @ everyone and @ here for group dms

### CODE REVIEW

File: src/plugins/onePingPerDM/index.ts [hunk 1]
```diff
@@ -4,20 +4,44 @@
...
+            { label: "User DMs", value: "user_dm" },
+            { label: "Group DMs", value: "group_dm" },
+        ]
+    },
+    allowMentions: {
+        type: OptionType.BOOLEAN,
+        description: "Receive audio pings for @mentions",
+        default: false,
+    },
+    "Allow @everyone": {
```
Comment (Vendicated) [inline src/plugins/onePingPerDM/index.ts:33]:
```suggestion
    allowEveryone: {
```

File: src/plugins/onePingPerDM/index.ts [hunk 2]
```diff
@@ -29,11 +53,19 @@ export default definePlugin({
...
-    isPrivateChannelRead(message: Message) {
+    isPrivateChannelRead(message: MessageJSON) {
         const channelType = ChannelStore.getChannel(message.channel_id)?.type;
         if (channelType !== ChannelType.DM && channelType !== ChannelType.GROUP_DM) {
             return false;
         }
+        if (
+            (channelType === ChannelType.DM && settings.store.channelToEffect === "group_dm") ||
+            (channelType === ChannelType.GROUP_DM && settings.store.channelToEffect === "user_dm") ||
+            (settings.store.allowMentions && message.mentions.find(m => m.id === UserStore.getCurrentUser().id)) ||
```
Comment (Vendicated) [inline src/plugins/onePingPerDM/index.ts:64]:
```suggestion
            (settings.store.allowMentions && message.mentions.some(m => m.id === UserStore.getCurrentUser().id)) ||
```

**Comment by ProffDea** [comment]
new changes tested and good to go

File: src/plugins/onePingPerDM/index.ts [hunk 3]
```diff
@@ -4,20 +4,44 @@
...
+import { ChannelStore, ReadStateStore, UserStore } from "@webpack/common";
+import { MessageJSON } from "discord-types/general";
 
 const enum ChannelType {
     DM = 1,
     GROUP_DM = 3
 }
 
+const settings = definePluginSettings({
+    channelToEffect: {
```
Comment (Vendicated) [inline src/plugins/onePingPerDM/index.ts:19]: you silly goober
```suggestion
    channelToAffect: {
```

**Pull request merged**